crates/l3/appchain-core-contract-client/src/interfaces/core_contract.rs

```diff
@@ -0,0 +1,29 @@
+use color_eyre::Result;
+use common::invoke_contract;
+use common::LocalWalletSignerMiddleware;
+use starknet_core::types::{FieldElement, InvokeTransactionResult};
+use std::sync::Arc;
+
+pub struct CoreContract<'a> {
+    client: Arc<&'a LocalWalletSignerMiddleware>,
```

Comment:
Can't you just store `&'a LocalWalletSignerMiddleware` here? Rather than the arc?